src/kvirc/kernel/KviIrcConnectionServerInfo.cpp

```diff
@@ -275,6 +273,566 @@ KviBasicIrcServerInfo::~KviBasicIrcServerInfo()
+		case 'v': return __tr2qs("v: Allows the ability to view OPER privileges via WHOIS"); break;
+		case 'x': return __tr2qs("x: Allows the ability to see netjoins"); break;
+		case 'z': return __tr2qs("z: Recipient for OPERWALL messages"); break;
+		case '0': return __tr2qs("0: Allows the ability to view OPER information"); break;
+		case '1': return __tr2qs("1: Allows the ability to view I-Lines and Y-Lines"); break;
+		case '2': return __tr2qs("2: Allows the ability to view D-Lines and K-Lines"); break;
+		case '3': return __tr2qs("3: Allows the ability to view Q-Lines and X-Lines"); break;
+		case '4': return __tr2qs("4: Allows the ability to use STATS T"); break;
+		case '5': return __tr2qs("5: Allows the ability to use STATS ?"); break;
+		case '9': return __tr2qs("9: Allows the use of TESTLINE"); break;
+		case '*': return __tr2qs("*: Allows the user to grant umodes they have access to to other people"); break;
+		case '@': return __tr2qs("@: allows the user to change their host to anything with the SETHOST command"); break;
+	}
+	return KviBasicIrcServerInfo::getUserModeDescription(mode);
+}
+
+////////////////
+// UMODE Requirements
+////////////////
+
+QChar KviBasicIrcServerInfo::getUserModeRequiement(QChar mode)
+{
+	switch(mode.unicode())
+	{
+		case 'O': return 'O'; case 'a': return 'a';
+
+		case 'c': case 'd': case 'f': case 'k':
+		case 'n': case 'o': case 's': case 'y':
+		case 'z': return 'o';
+	}
+	return 0;
+}
+
+QChar KviUnreal32IrcServerInfo::getUserModeRequiement(QChar mode)
+{
+	switch(mode.unicode())
+	{
+		case 'A': return 'A'; case 'C': return 'C';
+		case 'N': return 'N'; case 'O': return 'O';
+		case 'a': return 'a'; case 'r': return 'r';
+		case 't': return 't';
+
+		case 'S': case 'V': case 'x': case 'z':
+		return 1;
+
+		case 'F': case 'H': case 'W': case 'f':
+		case 'g': case 'h': case 'o': case 's':
+		case 'v': case 'w': return 'o';
+
+		case 'q': return 'a';
+	}
+	return 0;
+}
+
+QChar KviHybridServerInfo::getUserModeRequiement(QChar mode)
+{
+	switch(mode.unicode())
+	{
+		case 'a': return 'a';
+
+		case 'S': case 'W': case 'r': case 'x':
+		return 1;
+
+		case 'F': case 'H': case 'c': case 'd':
+		case 'e': case 'f': case 'j': case 'k':
+		case 'l': case 'n': case 'o': case 's':
+		case 'u': case 'y': return 'o';
+	}
+	return 0;
+}
+
+QChar KviCritenIrcServerInfo::getUserModeRequiement(QChar mode)
+{
+	switch(mode.unicode())
+	{
+		case 'O': return 'O';
+
+		case 'P': case 'S': case 'Z': case 'a':
+		case 'r': case 'x': return 1;
+
+		case 'c': case 'C': case 'd': case 'e':
+		case 'f': case 'F': case 'g': case 'm':
+		case 'n': case 'p': case 'W': case 'y':
+		case 'h': case 'o': return 'o';
+	}
+	return 0;
+}
+
+QChar KviBahamutIrcServerInfo::getUserModeRequiement(QChar mode)
+{
+	switch(mode.unicode())
+	{
+		case 'A': return 'A'; case 'O': return 'O';
+
+		case 'S': case 'W': case 'X': case 'a':
+		case 'r': case 'w': case 'x': return 1;
```

Comment:
As I said in my description, I return 0 for settable, a QChar for the mode required to set it, and a 1 for never settable. I couldn't pick a random non existent mode since projects like hyperion decide to use 59 modes, run out numbers, then switch to symbols. I'm open to better ideas, but but was the easiest implementation I could think of.